src/transformers/models/altclip/modeling_altclip.py

```diff
@@ -0,0 +1,1713 @@
+        # if isinstance(module, CLIPTextEmbeddings):
+        #     module.token_embedding.weight.data.normal_(mean=0.0, std=factor * 0.02)
+        #     module.position_embedding.weight.data.normal_(mean=0.0, std=factor * 0.02)
+        if isinstance(module, AltCLIPVisionEmbeddings):
+            factor = self.config.initializer_factor
+            nn.init.normal_(module.class_embedding, mean=0.0, std=module.embed_dim**-0.5 * factor)
+            nn.init.normal_(module.patch_embedding.weight, std=module.config.initializer_range * factor)
+            nn.init.normal_(module.position_embedding.weight, std=module.config.initializer_range * factor)
+        elif isinstance(module, AltCLIPAttention):
+            factor = self.config.initializer_factor
+            in_proj_std = (module.embed_dim**-0.5) * ((2 * module.config.num_hidden_layers) ** -0.5) * factor
+            out_proj_std = (module.embed_dim**-0.5) * factor
+            nn.init.normal_(module.q_proj.weight, std=in_proj_std)
+            nn.init.normal_(module.k_proj.weight, std=in_proj_std)
+            nn.init.normal_(module.v_proj.weight, std=in_proj_std)
+            nn.init.normal_(module.out_proj.weight, std=out_proj_std)
+        elif isinstance(module, AltCLIPMLP):
+            factor = self.config.initializer_factor
+            in_proj_std = (
+                (module.config.hidden_size**-0.5) * ((2 * module.config.num_hidden_layers) ** -0.5) * factor
+            )
+            fc_std = (2 * module.config.hidden_size) ** -0.5 * factor
+            nn.init.normal_(module.fc1.weight, std=fc_std)
+            nn.init.normal_(module.fc2.weight, std=in_proj_std)
+        elif isinstance(module, AltCLIPModel):
+            nn.init.normal_(
+                module.text_projection.weight,
+                std=module.text_embed_dim**-0.5 * self.config.initializer_factor,
+            )
+            nn.init.normal_(
+                module.visual_projection.weight,
+                std=module.vision_embed_dim**-0.5 * self.config.initializer_factor,
+            )
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        elif isinstance(module, nn.Linear):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_factor)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_factor)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, AltCLIPEncoder):
+            module.gradient_checkpointing = value
+        if isinstance(module, AltRobertaEncoder):
+            module.gradient_checkpointing = value
+
+
+# CLIP_->ALTCLIP_,CLIP->AltCLIP
```

Comment:
The copied from is missing here.
```suggestion
# Copied from transformers.models.clip.modeling_clip.CLIPVisionTransformer with CLIP_->ALTCLIP,CLIP->AltCLIP
```

Reply:
I got some problem here.

when I run `make fix-copies`, the rules below changed
```python
 @add_start_docstrings_to_model_forward(ALTCLIP_VISION_INPUTS_DOCSTRING)
```
to
```python
    @add_start_docstrings_to_model_forward(ALTAltCLIPVISION_INPUTS_DOCSTRING)
```

it seems like rename the attribute twice. How should i do ?

Reply:
I tried `CLIP_->ALTCLIP_`, it didn't work properly either. Look like it will execute two different rule indecently

Reply:
Ah yes. You should revert to how it was before then (the very long `Copied from`). Sorry I misguided you.